dsgrid/project.py

```diff
@@ -125,8 +164,80 @@ def unload_dataset(self, dataset_id):
         dataset.delete_views()
 
     def _iter_datasets(self):
-        for dataset in self.config.datasets:
+        for dataset in self.config.model.datasets:
             yield dataset
 
     def list_datasets(self):
         return [x.dataset_id for x in self._iter_datasets()]
+
+    @track_timing(timer_stats_collector)
+    def process_query(self, context: QueryContext):
+        self._build_filtered_record_ids_by_dimension_type(context)
+
+        dfs = []
+        for dataset_id in context.model.project.dataset_ids:
+            logger.info("Start processing query for dataset_id=%s", dataset_id)
+            dataset = self.get_dataset(dataset_id)
+            dfs.append(dataset.get_dataframe(context, self._config))
+            logger.info("Finished processing query for dataset_id=%s", dataset_id)
+
+        if not dfs:
+            logger.warning("No data matched %s", context.name)
+            return None
+
+        # All dataset columns need to be in the same order.
+        context.consolidate_dataset_metadata()
+        # TODO: Change this when we support long format.
+        pivoted_columns = sorted(context.get_pivoted_columns())
+        dim_columns = context.get_all_dimension_query_names()
+        time_columns = context.get_dimension_query_names(DimensionType.TIME)
+        dim_columns -= time_columns
+        for col in dim_columns:
```

Comment:
this could be refactored into context, sth to get cols that not neither pivoted or time